test/unit/specs/components/DecisioningEngine/createEvaluableRulesetPayload.spec.js

```diff
@@ -285,4 +285,218 @@ describe("DecisioningEngine:createEvaluableRulesetPayload", () => {
+            },
+            type: "group"
+          },
+          consequences: [
+            {
+              type: "item",
+              detail: {
+                schema: "https://ns.adobe.com/personalization/dom-action",
+                data: {
+                  selector:
+                    "HTML > BODY > DIV.offer:eq(0) > IMG:nth-of-type(1)",
+                  type: "setAttribute",
+                  content: {
+                    src: "img/demo-marketing-offer1-exp-A.png"
+                  },
+                  prehidingSelector:
+                    "HTML > BODY > DIV:nth-of-type(2) > IMG:nth-of-type(1)"
+                },
+                id: "79129ecf-6430-4fbd-955a-b4f1dfdaa6fe"
+              },
+              id: "79129ecf-6430-4fbd-955a-b4f1dfdaa6fe"
+            },
+            {
+              type: "item",
+              detail: {
+                schema: "https://ns.adobe.com/personalization/dom-action",
+                data: {
+                  selector:
+                    "HTML > BODY > DIV:nth-of-type(1) > H1:nth-of-type(1)",
+                  type: "setHtml",
+                  content: "Hello Treatment A!",
+                  prehidingSelector:
+                    "HTML > BODY > DIV:nth-of-type(1) > H1:nth-of-type(1)"
+                },
+                id: "10da709c-aa1a-40e5-84dd-966e2e8a1d5f"
+              },
+              id: "10da709c-aa1a-40e5-84dd-966e2e8a1d5f"
+            }
+          ]
+        }
+      ]
+    });
+
+    expect(
+      ruleset.execute({
+        "browser.name": "chrome",
+        "page.url": "https://www.adobe.com"
+      })
+    ).toEqual([
+      [
+        {
+          type: "item",
+          detail: {
+            schema: "https://ns.adobe.com/personalization/dom-action",
+            data: {
+              selector: "HTML > BODY > DIV.offer:eq(0) > IMG:nth-of-type(1)",
+              type: "setAttribute",
+              content: {
+                src: "img/demo-marketing-offer1-exp-A.png"
+              },
+              prehidingSelector:
+                "HTML > BODY > DIV:nth-of-type(2) > IMG:nth-of-type(1)"
+            },
+            id: "79129ecf-6430-4fbd-955a-b4f1dfdaa6fe"
+          },
+          id: "79129ecf-6430-4fbd-955a-b4f1dfdaa6fe"
+        },
+        {
+          type: "item",
+          detail: {
+            schema: "https://ns.adobe.com/personalization/dom-action",
+            data: {
+              selector: "HTML > BODY > DIV:nth-of-type(1) > H1:nth-of-type(1)",
+              type: "setHtml",
+              content: "Hello Treatment A!",
+              prehidingSelector:
+                "HTML > BODY > DIV:nth-of-type(1) > H1:nth-of-type(1)"
+            },
+            id: "10da709c-aa1a-40e5-84dd-966e2e8a1d5f"
+          },
+          id: "10da709c-aa1a-40e5-84dd-966e2e8a1d5f"
+        }
+      ]
+    ]);
+  });
+
+  it("evaluates global context and return empty consequence", () => {
+    const ruleset = RulesEngine({
```

Comment:
tests for RulesEngine alone should be in the rules engine repo instead of here.  we need to test that context is correctly generated and used when rulesets are evaluated in this repo.  We need to do that via the Decisioning Engine component code rather than bypassing it to test rule engine alone.